core/js/src/main/scala/com/avsystem/commons/serialization/nativejs/NativeJsonOutput.scala

```diff
@@ -0,0 +1,98 @@
+package com.avsystem.commons
+package serialization.nativejs
+
+import com.avsystem.commons.serialization._
+import com.avsystem.commons.serialization.json.RawJson
+
+import scala.scalajs.js
+import scala.scalajs.js.JSON
+
+final class NativeJsonOutput(
+  valueConsumer: js.Any => Unit,
+  options: NativeFormatOptions,
+) extends OutputAndSimpleOutput {
+
+  override def writeNull(): Unit =
+    valueConsumer(null)
+
+  override def writeString(str: String): Unit =
+    valueConsumer(str)
+
+  override def writeDouble(double: Double): Unit =
+    valueConsumer(double)
+
+  override def writeInt(int: Int): Unit =
+    valueConsumer(int)
+
+  override def writeLong(long: Long): Unit = options.longFormat match {
+    case NativeLongFormat.RawString => writeString(long.toString)
+    case NativeLongFormat.JsNumber => writeDouble(long.toDouble)
+    case NativeLongFormat.JsBigInt => writeRaw(js.BigInt(long.toString))
+  }
+
+  override def writeBigInt(bigInt: BigInt): Unit =
+    writeString(bigInt.toString)
+
+  override def writeBigDecimal(bigDecimal: BigDecimal): Unit =
+    writeString(bigDecimal.toString)
+
+  override def writeBoolean(boolean: Boolean): Unit =
+    valueConsumer(boolean)
+
+  override def writeList(): ListOutput =
+    new NativeJsonListOutput(valueConsumer, options)
+
+  override def writeObject(): ObjectOutput =
+    new NativeJsonObjectOutput(valueConsumer, options)
+
+  override def writeBinary(binary: Array[Byte]): Unit = {
+    val l = writeList()
+    binary.foreach(b => l.writeElement().writeSimple().writeInt(b))
```

Comment:
Writing bytes spawns a ton of short-living objects (each `writeElement()` creates a new output object). It seems that this works with a single object just as well?

Reply:
Changed to simple array conversion